### PR TITLE
Add files from GAX and remove Guava from surface

### PR DESCRIPTION
**Pull request by michaelbausor**

### CODE REVIEW

File: src/main/java/com/google/api/common/ApiClock.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2016, Google Inc.
```
Comment (michaelbausor) [inline src/main/java/com/google/api/common/ApiClock.java]: @garrettjonesgoogle Are these files "new"?

File: src/main/java/com/google/api/pathtemplate/PathTemplate.java [hunk 2]
```diff
@@ -69,7 +69,7 @@
  * <p>Here is an example for a template using simple variables:
  * <pre>
  *   PathTemplate template = PathTemplate.create("v1/shelves/{shelf}/books/{book}");
- *   assert template.match("v2/shelves"} == false
+ *   assert template.matches("v2/shelves") == false;
  *   Map&lt;String, String&gt; values = template.match("v1/shelves/s1/books/b1");
  *   assert values.equals(ImmutableMap.of("shelf", s1", "book", "b1");
```
Comment (michaelbausor) [inline src/main/java/com/google/api/pathtemplate/PathTemplate.java]: @garrettjonesgoogle should I remove these references also?

File: src/main/java/com/google/api/common/ApiClock.java (hunk 1 above)
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/common/ApiClock.java]: Hmm I think original copyrights is correct. They are just being moved from another repo.

File: src/main/java/com/google/api/pathtemplate/PathTemplate.java (hunk 2 above)
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/pathtemplate/PathTemplate.java]: yes I think so.

Comment (michaelbausor) [inline src/main/java/com/google/api/pathtemplate/PathTemplate.java]: done

**Comment by garrettjonesgoogle** [comment]
LGTM

File: src/main/java/com/google/api/common/ListenableFutureToApiFuture.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import com.google.common.util.concurrent.ForwardingListenableFuture.SimpleForwardingListenableFuture;
+import com.google.common.util.concurrent.ListenableFuture;
+
+/**
+ * INTERNAL USE ONLY. Adapter from Guava ListenableFuture to GAX ApiFuture.
+ */
+@InternalApi
+public class ListenableFutureToApiFuture<V> extends SimpleForwardingListenableFuture<V>
+    implements ApiFuture<V> {
+  public ListenableFutureToApiFuture(ListenableFuture<V> delegate) {
```
Comment (michaelbausor) [inline src/main/java/com/google/api/common/ListenableFutureToApiFuture.java]: @garrettjonesgoogle We have ListenableFuture in the surface here. Is this OK? I don't see how we can get rid of it, given the purpose of this class.

Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/common/ListenableFutureToApiFuture.java]: This is why it is marked `@InternalApi` :-)

**Pull request merged**